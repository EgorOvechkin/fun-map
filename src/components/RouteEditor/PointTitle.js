import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './styles/PointTitle.css';
import SortablePoint from './SortablePoint';

class PointTitle extends PureComponent {
  constructor(props) {
    super(props);
    this.deletePoint = () => this.props.deletePoint(this.props.index);
  }

  render() {
    return (
      <SortablePoint title={this.props.title} index={this.props.index} deletePoint={this.deletePoint} />
    );
  }
}

PointTitle.propTypes = {
  title: PropTypes.string,
  deletePoint: PropTypes.func
};

export default PointTitle;
