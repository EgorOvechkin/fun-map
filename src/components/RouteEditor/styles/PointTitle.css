.point-title-container {
  border: 1px solid;
  list-style-type: none;
  background: #fff;
  cursor: grab;
}

.point-delete-button {
  cursor: pointer;
}