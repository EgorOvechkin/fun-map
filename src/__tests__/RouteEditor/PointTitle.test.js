import React from 'react';
import PointTitle from '../../components/RouteEditor/PointTitle';
import {shallow} from 'enzyme';

it('renders point title', () => {
  const wrapper = shallow(<PointTitle index={1} title={'Point'} />);
  expect(wrapper).toMatchSnapshot();
});

