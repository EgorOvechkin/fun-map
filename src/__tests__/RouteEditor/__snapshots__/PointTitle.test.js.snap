// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders point title 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PointTitle
    index={1}
    title="Point"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "deletePoint": [Function],
      "index": 1,
      "title": "Point",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "deletePoint": [Function],
        "index": 1,
        "title": "Point",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
